dll/win32/lpk/lpk.c

```diff
@@ -81,16 +83,31 @@ LpkExtTextOut(
     /* Check if the string requires complex script processing and not a "glyph indices" array */
     if (ScriptIsComplex(lpString, uCount, SIC_COMPLEX) == S_OK && !(fuOptions & ETO_GLYPH_INDEX))
     {
+        reordered_str = HeapAlloc(GetProcessHeap(), 0, uCount * sizeof(WCHAR));
```

Comment:
I would suggest adding a comment that says, that the reordered string is somehow optional and is (or seems to be?) used as a fallback to be displayed in case we failed to obtain reordered glyphs that we would otherwise use to print the text.
And thus it's not needed to check for returned NULL pointer because BIDI_Reorder ignores in that case the `reordered_str` parameter.